### PR TITLE
CSP-1575 remove permissions endpoint

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.PR.Application/Common/SuccessCommandResult.cs
Comment (DevenKShah) [inline src/SFA.DAS.PR.Application/Common/SuccessCommandResult.cs:1]: Why is this removed?

File: src/SFA.DAS.PR.Application/Common/Validators/ProviderValidator.cs
Comment (DevenKShah) [inline src/SFA.DAS.PR.Application/Common/Validators/ProviderValidator.cs:1]: This is already covered by Shaun, can we reconcile the changes, a lot of these has been duplicated.

File: src/SFA.DAS.PR.Application/Common/SuccessCommandResult.cs
Comment (MarkFCain) [reply src/SFA.DAS.PR.Application/Common/SuccessCommandResult.cs:1]: there was also one in /Common/Commands/SuccessCommandResult.cs that is exactly the same, so making it coherent

File: src/SFA.DAS.PR.Application/Permissions/Commands/RemovePermissions/RemovePermissionsCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            return await RemoveAndAuditPermissions(accountProviderLegalEntity, command, cancellationToken);
+        }
+
+        return new ValidatedResponse<SuccessCommandResult>();
+    }
+
+    private async Task<ValidatedResponse<SuccessCommandResult>> RemoveAndAuditPermissions(AccountProviderLegalEntity accountProviderLegalEntity, RemovePermissionsCommand command, CancellationToken cancellationToken)
+    {
+        var operationsToRemove = accountProviderLegalEntity.Permissions.Select(permission => permission.Operation).ToList();
```
Comment (DevenKShah) [inline src/SFA.DAS.PR.Application/Permissions/Commands/RemovePermissions/RemovePermissionsCommandHandler.cs:37]: There is no need to convert this into a list, IEnumerable is serialisable I think

**Review by DevenKShah — DISMISSED** [review]
I have suggested some minor improvements

File: src/SFA.DAS.PR.Application/Permissions/Commands/RemovePermissions/RemovePermissionsCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        RemovePermissions(accountProviderLegalEntity.Permissions.ToList());
+
+        await DeletePermissionsAudit(command, PermissionAction.PermissionDeleted.ToString(), operationsToRemove, cancellationToken);
+        await _providerRelationshipsDataContext.SaveChangesAsync(cancellationToken);
+
+        return new ValidatedResponse<SuccessCommandResult>();
+    }
+
+    private void RemovePermissions(List<Permission>? permissions)
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Application/Permissions/Commands/RemovePermissions/RemovePermissionsCommandHandler.cs]: no reason, changed to non-niullable

File: src/SFA.DAS.PR.Application/Permissions/Commands/RemovePermissions/RemovePermissionsCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        }
+
+        return new ValidatedResponse<SuccessCommandResult>();
+    }
+
+    private async Task<ValidatedResponse<SuccessCommandResult>> RemoveAndAuditPermissions(AccountProviderLegalEntity accountProviderLegalEntity, RemovePermissionsCommand command, CancellationToken cancellationToken)
+    {
+        var operationsToRemove = accountProviderLegalEntity.Permissions.Select(permission => permission.Operation).ToList();
+
+        RemovePermissions(accountProviderLegalEntity.Permissions.ToList());
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.PR.Application/Permissions/Commands/RemovePermissions/RemovePermissionsCommandHandler.cs]: fixed

File: src/SFA.DAS.PR.Application/Permissions/Commands/RemovePermissions/RemovePermissionsCommandHandler.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.PR.Application/Permissions/Commands/RemovePermissions/RemovePermissionsCommandHandler.cs]: fixed

**Review by DevenKShah — APPROVED** [review]
